src/main/webapp/app/admin/legal/legal-document-update.component.ts

```diff
@@ -0,0 +1,138 @@
+    unsavedChanges = false;
+    isSaving = false;
+    @ViewChild(MarkdownEditorComponent, { static: false }) markdownEditor: MarkdownEditorComponent;
+
+    currentLanguage = this.defaultLanguage;
+    unsavedChangesWarning: NgbModalRef;
+    titleKey: string;
+
+    constructor(
+        private legalDocumentService: LegalDocumentService,
+        private modalService: NgbModal,
+        private route: ActivatedRoute,
+        private languageHelper: JhiLanguageHelper,
+        private changeDetectorRef: ChangeDetectorRef,
+    ) {}
+
+    ngOnInit() {
+        // Tap the URL to determine, if it's the imprint or the privacy statement
+        // we need the parent URL, because the imprint and privacy statement are children of the admin component and their path is specified there because they are lazy loaded
+        this.route.parent?.url
+            .pipe(
+                tap((segments) => {
+                    this.legalDocumentType = segments.some((segment) => segment.path === 'imprint') ? LegalDocumentType.IMPRINT : LegalDocumentType.PRIVACY_STATEMENT;
+                }),
+            )
+            .subscribe();
+        if (this.legalDocumentType === LegalDocumentType.IMPRINT) {
+            this.languageHelper.updateTitle('artemisApp.legal.imprint.updateImprint');
+            this.titleKey = 'artemisApp.legal.imprint.updateImprint';
+        } else {
+            this.languageHelper.updateTitle('artemisApp.legal.privacyStatement.updatePrivacyStatement');
+            this.titleKey = 'artemisApp.legal.privacyStatement.updatePrivacyStatement';
+        }
+        this.legalDocument = new LegalDocument(this.legalDocumentType, this.defaultLanguage);
+        this.getLegalDocumentForUpdate(this.legalDocumentType, this.defaultLanguage).subscribe((document) => {
+            this.legalDocument = document;
+        });
+    }
+
+    private getLegalDocumentForUpdate(type: LegalDocumentType, language: LegalDocumentLanguage): Observable<LegalDocument> {
+        if (type === LegalDocumentType.PRIVACY_STATEMENT) {
+            return this.legalDocumentService.getPrivacyStatementForUpdate(language);
+        } else {
+            return this.legalDocumentService.getImprintForUpdate(language);
+        }
+    }
+
+    updateLegalDocument() {
+        this.isSaving = true;
+        this.legalDocument.text = this.markdownEditor.markdown!;
+        if (this.legalDocumentType === LegalDocumentType.PRIVACY_STATEMENT) {
+            this.legalDocumentService.updatePrivacyStatement(this.legalDocument).subscribe((statement) => {
+                this.legalDocument = statement;
+                this.unsavedChanges = false;
+                this.isSaving = false;
+            });
+        } else {
+            this.legalDocumentService.updateImprint(this.legalDocument).subscribe((statement) => {
+                this.legalDocument = statement;
+                this.unsavedChanges = false;
+                this.isSaving = false;
+            });
```

Comment:
Maybe move the callback to a separate function / anonymous callback variable before this block to avoid duplication? Not strictly necessary